cmake/modules/RootMacros.cmake

```diff
@@ -1242,6 +1242,16 @@ function(ROOT_STANDARD_LIBRARY_PACKAGE libname)
     endif()
   endif()
 
+  foreach (h ${ARG_HEADERS})
+    if (EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/inc/${h}")
+      list(APPEND ARG_SOURCES "${CMAKE_CURRENT_SOURCE_DIR}/inc/${h}")
+    elseif (EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/v7/inc/${h}")
+      list(APPEND ARG_SOURCES "${CMAKE_CURRENT_SOURCE_DIR}/v7/inc/${h}")
+    else()
+      message(${h})
+    endif()
+  endforeach()
```

Comment:
Have a look at [`ROOT_GENERATE_DICTIONARY`](https://github.com/root-project/root/blob/7273b38f8aaa65dafb0d3e5516f6b15c4aebbedd/cmake/modules/RootMacros.cmake#L270-L710) for the logic to check if the header exists, it may make sense to something similar here or split that search logic into its own macro or function. `ARG_HEADERS` may actually contain more than just the headers for a given target, so I'm not sure we can use it verbatim to add headers to the targets. It should be ok if all targets use `NODEPHEADERS` properly, but it's probably worth checking anyway.

Reply:
I had a look and the logic seems overly complex to me. I would like to keep it simple. I improved the message in case the header is not found and on the current master, it is never triggered. Therefore this logic seems to suffice.

Reply:
Shouldn't we explicitly enumerate all sources when calling `ROOT_STANDARD_LIBRARY_PACKAGE` nowadays?

Reply:
YES! But people seem to have a different understanding of what a source is. In ROOT it seems people just add sources that need to be compiled. IMO at least all files containing code should be added.

For ROOT, that's a rather big change. I can try to do that for the libraries I will be working with, but I would like to have more support to move in this direction. Maybe we should discuss this in a team meeting.